<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Forward_InversekinematicsClass</class>
 <widget class="QWidget" name="Forward_InversekinematicsClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1509</width>
    <height>853</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Forward_Inversekinematics</string>
  </property>
  <widget class="QDoubleSpinBox" name="DampX">
   <property name="geometry">
    <rect>
     <x>980</x>
     <y>410</y>
     <width>81</width>
     <height>51</height>
    </rect>
   </property>
   <property name="accelerated">
    <bool>false</bool>
   </property>
   <property name="decimals">
    <number>15</number>
   </property>
   <property name="minimum">
    <double>-50.000000000000000</double>
   </property>
   <property name="maximum">
    <double>50.000000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_4">
   <property name="geometry">
    <rect>
     <x>1030</x>
     <y>610</y>
     <width>171</width>
     <height>51</height>
    </rect>
   </property>
   <property name="text">
    <string>Adding control demo</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="JointX">
   <property name="geometry">
    <rect>
     <x>980</x>
     <y>290</y>
     <width>81</width>
     <height>51</height>
    </rect>
   </property>
   <property name="accelerated">
    <bool>false</bool>
   </property>
   <property name="decimals">
    <number>15</number>
   </property>
   <property name="minimum">
    <double>-50.000000000000000</double>
   </property>
   <property name="maximum">
    <double>50.000000000000000</double>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="DampZ">
   <property name="geometry">
    <rect>
     <x>1160</x>
     <y>410</y>
     <width>81</width>
     <height>51</height>
    </rect>
   </property>
   <property name="accelerated">
    <bool>false</bool>
   </property>
   <property name="decimals">
    <number>15</number>
   </property>
   <property name="minimum">
    <double>-50.000000000000000</double>
   </property>
   <property name="maximum">
    <double>50.000000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>860</x>
     <y>630</y>
     <width>161</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>PROBLEM NOT FIXEDâ†’</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>910</x>
     <y>10</y>
     <width>81</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Joint Name</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>890</x>
     <y>430</y>
     <width>72</width>
     <height>15</height>
    </rect>
   </property>
   <property name="text">
    <string>Damp IK</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_3">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>91</width>
     <height>51</height>
    </rect>
   </property>
   <property name="text">
    <string>Stop/Start</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="lineEdit">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>790</y>
     <width>461</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>INPUT your path here to load</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="DampY">
   <property name="geometry">
    <rect>
     <x>1070</x>
     <y>410</y>
     <width>81</width>
     <height>51</height>
    </rect>
   </property>
   <property name="accelerated">
    <bool>false</bool>
   </property>
   <property name="decimals">
    <number>15</number>
   </property>
   <property name="minimum">
    <double>-50.000000000000000</double>
   </property>
   <property name="maximum">
    <double>50.000000000000000</double>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>1010</x>
     <y>380</y>
     <width>201</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>X          Y          Z</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>530</x>
     <y>790</y>
     <width>131</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>WriteFile</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>310</y>
     <width>91</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Regular IK</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="JointZ">
   <property name="geometry">
    <rect>
     <x>1160</x>
     <y>290</y>
     <width>81</width>
     <height>51</height>
    </rect>
   </property>
   <property name="accelerated">
    <bool>false</bool>
   </property>
   <property name="decimals">
    <number>15</number>
   </property>
   <property name="minimum">
    <double>-50.000000000000000</double>
   </property>
   <property name="maximum">
    <double>50.000000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_2">
   <property name="geometry">
    <rect>
     <x>1040</x>
     <y>520</y>
     <width>151</width>
     <height>71</height>
    </rect>
   </property>
   <property name="text">
    <string>Commence 2 Joints</string>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>1010</x>
     <y>710</y>
     <width>211</width>
     <height>121</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QSlider" name="RootX">
      <property name="minimum">
       <number>-50</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSlider" name="RootY">
      <property name="minimum">
       <number>-50</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="sliderPosition">
       <number>0</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSlider" name="RootZ">
      <property name="minimum">
       <number>-50</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>1010</x>
     <y>260</y>
     <width>201</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>X          Y          Z</string>
   </property>
  </widget>
  <widget class="QComboBox" name="comboBox">
   <property name="geometry">
    <rect>
     <x>1010</x>
     <y>0</y>
     <width>261</width>
     <height>41</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>Hips</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LHipJoint</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LeftUpLeg</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LeftLeg</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LeftFoot</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LeftToeBase</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RHipJoint</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RightUpLeg</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RightLeg</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RightFoot</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RightToeBase</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LowerBack</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Spine</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Spine1</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Neck</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Neck1</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Head</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LeftShoulder</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LeftArm</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LeftForeArm</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LeftHand</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LeftFingerBase</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LeftHandIndex1</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>LThumb</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RightShoulder</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RightArm</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RightForeArm</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RightHand</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RightFingerBase</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RightHandIndex1</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>RThumb</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>1050</x>
     <y>670</y>
     <width>151</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Root Translation</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="JointY">
   <property name="geometry">
    <rect>
     <x>1070</x>
     <y>290</y>
     <width>81</width>
     <height>51</height>
    </rect>
   </property>
   <property name="accelerated">
    <bool>false</bool>
   </property>
   <property name="decimals">
    <number>15</number>
   </property>
   <property name="minimum">
    <double>-50.000000000000000</double>
   </property>
   <property name="maximum">
    <double>50.000000000000000</double>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="Forward_Inversekinematics.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>addControlIK()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1069</x>
     <y>640</y>
    </hint>
    <hint type="destinationlabel">
     <x>868</x>
     <y>601</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>Stop_Start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>IK2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1078</x>
     <y>558</y>
    </hint>
    <hint type="destinationlabel">
     <x>963</x>
     <y>544</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>JointX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>changePositionX(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>995</x>
     <y>316</y>
    </hint>
    <hint type="destinationlabel">
     <x>832</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DampX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>changePositionX_Damp(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1014</x>
     <y>435</y>
    </hint>
    <hint type="destinationlabel">
     <x>769</x>
     <y>473</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>JointY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>changePositionY(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1112</x>
     <y>315</y>
    </hint>
    <hint type="destinationlabel">
     <x>1097</x>
     <y>175</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DampY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>changePositionY_Damp(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1097</x>
     <y>434</y>
    </hint>
    <hint type="destinationlabel">
     <x>1100</x>
     <y>482</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>JointZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>changePositionZ(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1195</x>
     <y>307</y>
    </hint>
    <hint type="destinationlabel">
     <x>1218</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DampZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>changePositionZ_Damp(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1179</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>1281</x>
     <y>443</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>activated(QString)</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>getJoint(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1070</x>
     <y>24</y>
    </hint>
    <hint type="destinationlabel">
     <x>933</x>
     <y>94</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>loadFile(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>798</y>
    </hint>
    <hint type="destinationlabel">
     <x>223</x>
     <y>683</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RootX</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>rootTranslateX(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1116</x>
     <y>732</y>
    </hint>
    <hint type="destinationlabel">
     <x>909</x>
     <y>721</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RootY</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>rootTranslateY(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1103</x>
     <y>767</y>
    </hint>
    <hint type="destinationlabel">
     <x>868</x>
     <y>772</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RootZ</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>rootTranslateZ(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1091</x>
     <y>808</y>
    </hint>
    <hint type="destinationlabel">
     <x>875</x>
     <y>817</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Forward_InversekinematicsClass</receiver>
   <slot>writeFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>589</x>
     <y>801</y>
    </hint>
    <hint type="destinationlabel">
     <x>561</x>
     <y>645</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addControlIK()</slot>
  <slot>Stop_Start()</slot>
  <slot>IK2()</slot>
  <slot>changePositionX(double)</slot>
  <slot>changePositionX_Damp(double)</slot>
  <slot>changePositionY(double)</slot>
  <slot>changePositionY_Damp(double)</slot>
  <slot>changePositionZ(double)</slot>
  <slot>changePositionZ_Damp(double)</slot>
  <slot>getJoint(QString)</slot>
  <slot>loadFile(QString)</slot>
  <slot>rootTranslateX(int)</slot>
  <slot>rootTranslateY(int)</slot>
  <slot>rootTranslateZ(int)</slot>
  <slot>writeFile()</slot>
 </slots>
</ui>
